test/tools/yulInterpreter/Interpreter.cpp

```diff
@@ -32,12 +32,33 @@
 #include <libdevcore/FixedHash.h>
 
 #include <boost/range/adaptor/reversed.hpp>
+#include <boost/algorithm/cxx11/all_of.hpp>
+
+#include <ostream>
 
 using namespace std;
 using namespace dev;
 using namespace yul;
 using namespace yul::test;
 
+void InterpreterState::dumpTraceAndState(ostream& _out) const
+{
+	_out << "Trace:" << endl;
+	for (auto const& line: trace)
+		_out << "  " << line << endl;
```

Comment:
We may have to separate out `dumpTrace` and `dumpState`. The differential fuzzer will only be comparing pre and post optimized states because traces may (correctly) differ pre and post memory optimization.

Reply:
Note to self: `dumpState` needs to replace the highlighted lines here

https://github.com/ethereum/solidity/blob/cf1c236d4945e4937ba5c6848f2ef6ee69e9ae4c/test/tools/ossfuzz/yulFuzzerCommon.cpp#L38-L40

Reply:
Hm, I thought the idea was to modify the trace such that it does not contain any memory information anymore?

Reply:
Although we should include e.g. the hash of the calldata for a `call` opcode.

Reply:
> Hm, I thought the idea was to modify the trace such that it does not contain any memory information anymore?

Ah, I see. I think I misunderstood the overall design. Indeed, in that case it makes sense to remove any memory information in the trace, relying on the state dump. In this case, please ignore my comment regarding separating trace and state dumps :-)

Reply:
Note to self: This PR closes #6808 as well.